### ISSUE TITLE
Create `new structures:AdapterType`

### ISSUE THREAD

**Issue by cdmgtri** [niemopen/niem-naming-design-rules#4]
Migrated from NIEM/NIEM-NDR#115

Create a `structures:AdapterType` for adapter types to extend, rather than including the one-off attribute `appinfo:externalAdapterTypeIndicator` in type declarations.

This would be consistent with the way association, augmentation, and metadata types are defined:

Type | Rule
--- | ---
association types | Extend `structures:AssociationType` or a derivative
augmentation types | Extend `structures:AugmentationType`
metadata types | Extend `structures:MetadataType`
object types | Extend `structures:ObjectType` or a derivative
adapter types | Extend **`structures:ObjectType`** and add **`appinfo:externalAdapterTypeIndicator="true"`**

**Current**

```xml
<xs:complexType name="GeometryType" appinfo:externalAdapterTypeIndicator="true">
  <xs:complexContent>
    <xs:extension base="structures:ObjectType">
      <xs:sequence>
        <xs:element ref="gml:AbstractGeometry" minOccurs="1" maxOccurs="1"/>
      </xs:sequence>
    </xs:extension>
  </xs:complexContent>
</xs:complexType>
```

**Proposed**

```xml
<xs:complexType name="GeometryType">
  <xs:complexContent>
    <xs:extension base="structures:AdapterType">
      <xs:sequence>
        <xs:element ref="gml:AbstractGeometry" minOccurs="1" maxOccurs="1"/>
      </xs:sequence>
    </xs:extension>
  </xs:complexContent>
</xs:complexType>
```

#resolved Folks are fine with this.

**Comment by cdmgtri** [comment]
NDR adjustments:

- [ ] Rule [10-8](https://niem.github.io/NIEM-NDR/v5.0/niem-ndr.html#rule_10-8): External adapter type has indicator
- [ ] Rule [10-9](https://niem.github.io/NIEM-NDR/v5.0/niem-ndr.html#rule_10-9): Structure of external adapter type definition follows pattern
- [ ] Rule [10-71](https://niem.github.io/NIEM-NDR/v5.0/niem-ndr.html#rule_10-71): External adapter type indicator annotates complex type

**Comment by cdmgtri** [comment]
Model changes merged into the dev branch for 6.0.
